tempesta_fw/t/functional/helpers/be.py

```diff
@@ -10,113 +10,107 @@
-        BTW, that may be fixed by exporting state of Temepsta FW via debugfs.
-        """
-        got_connection = self.accept_event.wait(timeout)
-        if (not got_connection):
-            self.shutdown()
-            msg = ("No connection from Tempesta FW (backend: {0}, timeout: {1})"
-                   .format(self.server_address, timeout))
-            raise Exception(msg)
-    
+	kwargs = dict(poll_interval = 0.05)
+	Thread.__init__(self, target=self.serve_forever, kwargs=kwargs)
+	self.start()
+	# Synchronize with Tempesta FW.
+	if (wait_tfw_timeout_sec):
+		self.wait_for_tfw(wait_tfw_timeout_sec)
+
+	def wait_for_tfw(self, timeout):
+#        Sleep until a first accepted connection (presumably from Tempesta FW).
+
+#	FIXME: race conditions possible: after the connection is established,
+ #       the Tempesta FW must add the server to a load-balancing scheduler and
+#        so on, so there is a time span, when the Tempesta is not yet ready to
+#        forward incoming requests to the connected back-end server. At this
+#        point we just hope that this time span is negligible.
+#        BTW, that may be fixed by exporting state of Temepsta FW via debugfs.
+		got_connection = self.accept_event.wait(timeout)
+	if (not got_connection):
+		self.shutdown()
+		msg = ("No connection from Tempesta FW (backend: {0}, timeout: {1})"
+		.format(self.server_address, timeout))
+		raise Exception(msg)
+
     # get_request() calls accept() that blocks until the first connection.
     # We just inject a synchronization with wait_for_tfw() there.
-    def get_request(self):
-        ret = super().get_request()
-        self.accept_event.set()
-        return ret
-
-class BackendHTTPRequestHandler(BaseHTTPRequestHandler):
-    """
-    A wrapper for BackendHTTPServer.backend_callback.
-    The class simply pushes HTTP requests to the callback, and then builds
-    responses from data returned by the callback.  
-    
-    That is done for simplicity. It is easier to code a single callback function
-    than a whole handler class. We have to code one in every test, and we don't
-    need much power in tests code, so we prefer a function over a class.
-    """
-    def _handle_req_with_cb(self):
-        """
-        Pass HTTP request to backend_callback, send a response containing data
-        returned by the callback.
-        """
+	def get_request(self):
+		ret = super().get_request()
+	self.accept_event.set()
+	return ret
+
+	class BackendHTTPRequestHandler(BaseHTTPRequestHandler):
+
+#    A wrapper for BackendHTTPServer.backend_callback.
+#    The class simply pushes HTTP requests to the callback, and then builds
+#    responses from data returned by the callback.
+
+#    That is done for simplicity. It is easier to code a single callback function
+#    than a whole handler class. We have to code one in every test, and we don't
+#    need much power in tests code, so we prefer a function over a class.
```

Comment:
If you need identation for comments, please place it before the comment symbol. I.e

                # some comment

instead of

        #        some comment